### PR TITLE
Feature custom names record

### PR DESCRIPTION
**Pull request by andrea-escartin**
Allow to have custom channel names in recorded csv files

### CODE REVIEW

File: exploredesktop/modules/settings_module.py [hunk 1]
```diff
@@ -316,3 +359,326 @@ def enable_settings(self, enable=True) -> None:
...
+    def setModelData(self, editor, model, index):
+        """Set data in the model
+        """
+        if model.columns[index.column()]['editor'] == 'combobox':
+            model.setData(index, editor.currentText(), Qt.EditRole)
+            # model.reset()
+        QStyledItemDelegate.setModelData(self, editor, model, index)
+
+
+class ConfigTableModel(QAbstractTableModel, BaseModel):
```
Comment (SonjaSt) [inline exploredesktop/modules/settings_module.py:481]: The description for the class is missing.

File: exploredesktop/modules/settings_module.py [hunk 2]
```diff
@@ -316,3 +359,326 @@ def enable_settings(self, enable=True) -> None:
...
+            d = str(property["input"])
+        elif property_name == 'enable':
+            d = str(property["enable"])
+        elif property_name == 'name':
+            d = str(property["name"])
+        elif property_name == 'type':
+            d = str(property["type"])
+        else:
+            d = None
+        return d
```
Comment (SonjaSt) [inline exploredesktop/modules/settings_module.py:668]: see [comment on previous PR](https://github.com/Mentalab-hub/explore-desktop/pull/84/files#r980000769)

File: exploredesktop/modules/imp_module.py [hunk 3]
```diff
@@ -41,9 +41,9 @@ def display_default_imp(self) -> None:
         chan_dict = self.model.explorer.get_chan_dict()
         n_chan = self.model.explorer.n_active_chan
         pos = np.array([[0 + i * 3, 0] for i in range(n_chan)], dtype=float)
-        texts = [f"{key}\nNA" for key in chan_dict if chan_dict[key] == 1]
+        # TODO: show all channels, gray if disabled?
+        texts = [f"{one_chan_dict['name']}\nNA" for one_chan_dict in chan_dict if one_chan_dict['enable'] == 1]
```
Comment (SonjaSt) [inline exploredesktop/modules/imp_module.py:45]: `== 1` can be left out.

File: exploredesktop/modules/ui/dialog_recording_settings.py [hunk 4]
```diff
@@ -8,61 +8,25 @@
...
+    QVBoxLayout)
 
 class Ui_Dialog(object):
     def setupUi(self, Dialog):
         if not Dialog.objectName():
             Dialog.setObjectName(u"Dialog")
-        Dialog.resize(400, 237)
-        self.verticalLayout_2 = QVBoxLayout(Dialog)
-        self.verticalLayout_2.setObjectName(u"verticalLayout_2")
+        Dialog.resize(460, 285)
```
[the remark this answers is not in the document]
Comment (andrea-escartin) [reply exploredesktop/modules/ui/dialog_recording_settings.py:27]: This file is generated by QtCreator, we could change it but then every time we make a change we would need to change it manually again. I'll leave it like this for now, and I'll check if there is a better solution longterm

**Review by salman2135 — APPROVED** [review]

**Pull request merged**